source/Detector/Amptek/SGM/AmptekCommandManagerSGM.cpp

```diff
@@ -0,0 +1,218 @@
+#include "AmptekCommandManagerSGM.h"
+
+/// ===================== implementation of the static instance and methods ==================
+AmptekCommandManagerSGM* AmptekCommandManagerSGM::instance_ = 0;
+AmptekCommandManagerSGM* AmptekCommandManagerSGM::sddCommands(){
+	if(instance_ == 0)
+		instance_ = new AmptekCommandManagerSGM();
+
+	return instance_;
+}
+
+void AmptekCommandManagerSGM::releaseSDDCommands(){
+	if(instance_) {
+		delete instance_;
+		instance_ = 0;
+	}
+}
+
+/// ===================== implementation of the public methods ==================
+AmptekCommandManagerSGM::~AmptekCommandManagerSGM()
+{
+}
+
+//QString AmptekCommandManagerSGM::requestStatusPacketHex() const{
```

Comment:
These commented codes/functions will be deleted before merge.

I created a function ```requestDataPacket()``` to replace all those functions. I keep those replaced codes here for the convenience of reviewers to understand the functionality of ```requestDataPacket()```.